common/src/main/java/com/ericsson/bss/cassandra/ecaudit/common/chronicle/AuditRecordWriteMarshallable.java

```diff
@@ -32,20 +32,16 @@ public AuditRecordWriteMarshallable(AuditRecord auditRecord)
     @Override
     public void writeMarshallable(@NotNull WireOut wire)
     {
+        boolean batchRecord = auditRecord.getBatchId().isPresent();
+        String type = batchRecord ? WireTags.VALUE_TYPE_BATCH_ENTRY : WireTags.VALUE_TYPE_SINGLE_ENTRY;
+
         wire.write(WireTags.KEY_VERSION).int16(WireTags.VALUE_VERSION_CURRENT);
-        if (auditRecord.getBatchId().isPresent())
-        {
-            wire.write(WireTags.KEY_TYPE).text(WireTags.VALUE_TYPE_BATCH_ENTRY);
-        }
-        else
-        {
-            wire.write(WireTags.KEY_TYPE).text(WireTags.VALUE_TYPE_SINGLE_ENTRY);
-        }
+        wire.write(WireTags.KEY_TYPE).text(type);
         wire.write(WireTags.KEY_TIMESTAMP).int64(auditRecord.getTimestamp());
         wire.write(WireTags.KEY_CLIENT).bytes(auditRecord.getClientAddress().getAddress());
         wire.write(WireTags.KEY_COORDINATOR).bytes(auditRecord.getCoordinatorAddress().getAddress());
         wire.write(WireTags.KEY_USER).text(auditRecord.getUser());
-        if (auditRecord.getBatchId().isPresent())
+        if (batchRecord)
```

Comment:
+1

Reply:
I was not really thinking about the batch-id writing, just wanted to remove the if-else-statement from the "middle of" the wire writing... but it is a bit nicer having the ifPresent there!

What I think would be nice is to treat the wire like a builder pattern:
wire.write(tag1).text(value1)
      .write(tag2).text(value2)
      .write(tag3).butes(value3)
      ...

But the optional parameter (batchId) prevents this :-/

Maybe if the optional parameters would have been added at the end, it could have been somewhat nicer...

Reply:
We could move the optional parameter to the end. But as we discussed, there will be more optional parameters rather soon so won't really make any difference.

I'd prefer the ifPresent() approach.